lib/rubocop/cop/style/empty_line_after_guard_clause.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop enforces empty line after guard clause
+      #
+      # @example
+      #
+      #   # bad
+      #   def foo
+      #     return if need_return?
+      #     bar
+      #   end
+      #
+      #   # good
+      #   def foo
+      #     return if need_return?
+      #
+      #     bar
+      #   end
+      #
+      #   # good
+      #   def foo
+      #     if something?
+      #       do_something
+      #       return if need_return?
+      #     end
+      #   end
+      class EmptyLineAfterGuardClause < Cop
+        MSG = 'Add empty line after guard clause.'.freeze
+
+        def on_if(node)
+          return unless contains_guard_clause?(node)
+
+          if node.parent
+            return if node.parent.single_line?
+            return if next_sibling_guard_clause?(node)
+          end
+
+          return if next_line_empty?(node)
+
+          add_offense(node)
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            corrector.insert_after(node.source_range, "\n")
```

Comment:
This will do unexpected automatic correction when source comment is written at end of line. For example, in the following cases.

```ruby
return if foo? # A comment
bar
```

Probably using `node.source_range` would not be suitable for computing end of line.

Reply:
@koic fixed